src/Caller.sol

```diff
@@ -204,6 +204,6 @@ contract Caller is EIP712("Caller", "1"), ERC2771Context(address(this)) {
         returns (bytes memory returnData)
     {
         // Encode the message sender as per ERC-2771
-        return Address.functionCallWithValue(to, abi.encodePacked(data, sender), value);
+        return Address.functionCallWithValue(to, bytes.concat(data, bytes20(sender)), value);
```

Comment:
interesting. Is the concat cheaper?
Why the bytes20 cast also cheaper?

Reply:
No, it's the same, it's just that `abi.encodePacked` is now deprecated in Solidity in favor of `bytes.concat`.